fix go vet complaint on copy of lock value in tailchonk_test.go

### What is the issue?

Running "make check" on MacOS results in a go vet error:

```
./tool/go vet ./...
# [tailscale.com/tka](http://tailscale.com/tka)
tka/tailchonk_test.go:664:15: literal copies lock value from (*c.Chonk().(*Mem)): [tailscale.com/tka.Mem](http://tailscale.com/tka.Mem) contains sync.RWMutex

```
### Steps to reproduce

_No response_

### Are there any recent changes that introduced the issue?

_No response_

### OS

macOS

### OS version

_No response_

### Tailscale version

_No response_

### Other software

_No response_

### Bug report

_No response_

--- a/tka/tailchonk_test.go
+++ b/tka/tailchonk_test.go
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 	"testing"
 	"time"
 
@@ -620,6 +621,14 @@ func (c *compactingChonkFake) PurgeAUMs(hashes []AUMHash) error {
 	return nil
 }
 
+// Avoid go vet complaining about copying a lock value
+func cloneMem(src, dst *Mem) {
+	dst.l = sync.RWMutex{}
+	dst.aums = src.aums
+	dst.parentIndex = src.parentIndex
+	dst.lastActiveAncestor = src.lastActiveAncestor
+}
+
 func TestCompact(t *testing.T) {
 	fakeState := &State{
 		Keys:               []Key{{Kind: Key25519, Votes: 1}},
@@ -661,11 +670,12 @@ func TestCompact(t *testing.T) {
     `, optTemplate("checkpoint", AUM{MessageKind: AUMCheckpoint, State: fakeState}))
 
 	storage := &compactingChonkFake{
-		Mem:        (*c.Chonk().(*Mem)),
 		aumAge:     map[AUMHash]time.Time{(c.AUMHashes["F1"]): time.Now()},
 		t:          t,
 		wantDelete: []AUMHash{c.AUMHashes["A"], c.AUMHashes["B"], c.AUMHashes["OLD"]},
 	}
+
+	cloneMem(c.Chonk().(*Mem), &storage.Mem)
 
 	lastActiveAncestor, err := Compact(storage, c.AUMHashes["H"], CompactionOptions{MinChain: 2, MinAge: time.Hour})
 	if err != nil {

